vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -401,6 +403,13 @@ private void sendFile(RoutingContext context, String file, FileProps fileProps)
           // guess content type
           String contentType = MimeMapping.getMimeTypeForFilename(file);
           if (contentType != null) {
+            int li = file.lastIndexOf(46);
+            if (li != -1 && li != file.length() - 1) {
+              String ext = file.substring(li + 1, file.length());
+              if(compressedMediaTypes.contains(contentType) || compressedFileSuffixes.contains(ext)) {
```

Comment:
This check should not be nested in `if (contentType != null)`. It should apply even when we can't guess the `Content-Type`.